ext/crypto/lib.rs

```diff
@@ -1402,7 +1468,73 @@ pub async fn op_crypto_import_key(
             modulus_length: Some(private_key.modulus.as_bytes().len() * 8),
           })
         }
-        // TODO(@littledivy): spki
+        KeyFormat::Spki => {
+          let hash = args
+            .hash
+            .ok_or_else(|| type_error("Missing argument hash".to_string()))?;
+
+          // 2-3.
+          let pk_info =
+            spki::SubjectPublicKeyInfo::from_der(data).map_err(|e| {
+              custom_error("DOMExceptionOperationError", e.to_string())
+            })?;
+
+          // 4-5.
+          let alg = pk_info.algorithm.oid;
+
+          // 6.
+          let pk_hash = match alg {
+            // rsaEncryption
+            RSA_ENCRYPTION_OID => None,
+            // sha1WithRSAEncryption
+            SHA1_RSA_ENCRYPTION_OID => Some(CryptoHash::Sha1),
+            // sha256WithRSAEncryption
+            SHA256_RSA_ENCRYPTION_OID => Some(CryptoHash::Sha256),
+            // sha384WithRSAEncryption
+            SHA384_RSA_ENCRYPTION_OID => Some(CryptoHash::Sha384),
+            // sha512WithRSAEncryption
+            SHA512_RSA_ENCRYPTION_OID => Some(CryptoHash::Sha512),
```

Comment:
This is incorrect as per spec. The OID is `id-RSASSA-PSS` and hash algorithm is part of `RSASSA-PSS-params`. See implementation for pkcs8 above